custom/icds_reports/migrations/sql_templates/create_functions.sql

```diff
@@ -2146,3 +2146,74 @@ BEGIN
+        'CASE WHEN child_health_monthly.current_month_nutrition_status = ' || quote_literal('severely_underweight') || ' THEN 1' ||
+            'WHEN child_health_monthly.current_month_nutrition_status = ' || quote_literal('moderately_underweight') || ' THEN 2' ||
+            'WHEN child_health_monthly.current_month_nutrition_status = ' || quote_literal('normal') || ' THEN 3' ||
+            'ELSE 4 END AS current_month_nutrition_status_sort,' ||
+        'CASE WHEN child_health_monthly.current_month_stunting = ' || quote_literal('severe') || ' THEN 1' ||
+            'WHEN child_health_monthly.current_month_stunting = ' || quote_literal('moderate') || ' THEN 2' ||
+            'WHEN child_health_monthly.current_month_stunting = ' || quote_literal('normal') || ' THEN 3' ||
+            'ELSE 4 END AS current_month_stunting_sort,' ||
+        'CASE WHEN child_health_monthly.current_month_wasting = ' || quote_literal('severe') || ' THEN 1' ||
+            'WHEN child_health_monthly.current_month_wasting = ' || quote_literal('moderate') || '  THEN 2' ||
+            'WHEN child_health_monthly.current_month_wasting = ' || quote_literal('normal') || ' THEN 3' ||
+            'ELSE 4 END AS current_month_wasting_sort,' ||
+        'CASE ' ||
+          'WHEN child_health_monthly.zscore_grading_hfa = 1 AND child_health_monthly.zscore_grading_hfa_recorded_in_month = 1 THEN ' || quote_literal('severe') || ' ' ||
+          'WHEN child_health_monthly.zscore_grading_hfa = 2 AND child_health_monthly.zscore_grading_hfa_recorded_in_month = 1 THEN ' || quote_literal('moderate') || ' ' ||
+          'WHEN child_health_monthly.zscore_grading_hfa = 3 AND child_health_monthly.zscore_grading_hfa_recorded_in_month = 1 THEN ' || quote_literal('normal') || ' ' ||
+          'ELSE ' || quote_literal('') || ' END AS current_month_stunting_v2, ' ||
+        'CASE ' ||
+          'WHEN (child_health_monthly.zscore_grading_wfh = 1 AND child_health_monthly.zscore_grading_wfh_recorded_in_month = 1) OR (child_health_monthly.muac_grading = 1 AND child_health_monthly.muac_grading_recorded_in_month = 1) THEN ' || quote_literal('severe') || ' ' ||
+          'WHEN (child_health_monthly.zscore_grading_wfh = 2 AND child_health_monthly.zscore_grading_wfh_recorded_in_month = 1) OR (child_health_monthly.muac_grading = 2 AND child_health_monthly.muac_grading_recorded_in_month = 2) THEN ' || quote_literal('moderate') || ' ' ||
+          'WHEN (child_health_monthly.zscore_grading_wfh = 3 AND child_health_monthly.zscore_grading_wfh_recorded_in_month = 1) OR (child_health_monthly.muac_grading = 3 AND child_health_monthly.muac_grading_recorded_in_month = 3) THEN ' || quote_literal('normal') || ' ' ||
+          'ELSE '' || quote_literal('''') || '' END AS current_month_wasting_v2, ' ||
+        'CASE ' ||
+          'WHEN child_health_monthly.zscore_grading_hfa = 1 AND child_health_monthly.zscore_grading_hfa_recorded_in_month = 1 THEN 1 ' ||
```

Comment:
I think these could be simplified like `CASE WHEN child_health_monthly.zscore_grading_hfa_recorded_in_month = 1 THEN child_health_monthly.zscore_grading_hfa ELSE 4` similar for the other sorts columns

Reply:
@emord is there a possibility that in the ```zscore_grading_hfa``` we can have other value than 1,2 or 3?

Reply:
they can be 1, 2, 3 or 0 (if it hasn't been recorded) https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/ucr/data_sources/dashboard/dashboard_growth_monitoring.json#L233-L251